Content.Server/Ghost/GhostSystem.cs

```diff
@@ -399,8 +400,12 @@ private void OnToggleGhostVisibilityToAll(ToggleGhostVisibilityToAllEvent ev)
         public void MakeVisible(bool visible)
         {
             var entityQuery = EntityQueryEnumerator<GhostComponent, VisibilityComponent>();
-            while (entityQuery.MoveNext(out var uid, out _, out var vis))
+            while (entityQuery.MoveNext(out var uid, out var ghostComp, out var vis))
             {
+                // Don't show Revenants, best way to check for admin ghost is if they can interact
+                if (ghostComp.CanGhostInteract || HasComp<RevenantComponent>(uid))
+                    continue;
```

Comment:
Don't put specific component onto generic system, bad.

Almost every fork is going to have ghostsystem but almost none of them going to have revenantsystem.

Reply:
Eventbus is good, bool property, anything else that doesn't tie an antag to ghosts. Ghosts just handle state changes to / from another entity when a mob dies.

Reply:
So don't check for ghostcomp or is it specifically rev? IIRC there was some discussion about using a tag in discord last night. Think that might be better instead.